From 4cd04bc6f50de461a71228b8b537dd1e4f005d14 Mon Sep 17 00:00:00 2001
From: Pzqqt <821026875@qq.com>
Date: Fri, 12 Sep 2025 01:54:54 +0700
Subject: [PATCH] Revert "drm: Add valid clones check"

On HyperOS, this caused the screen to frequently flicker as the display refreshed.

This reverts commit ffb55ddf26850a8af666d9c04086ad9b34b6a8a4.

Change-Id: I780a043dce2edfa6096acea214acb3949e699146
---
 drivers/gpu/drm/drm_atomic_helper.c | 28 ----------------------------
 1 file changed, 28 deletions(-)

diff --git a/drivers/gpu/drm/drm_atomic_helper.c b/drivers/gpu/drm/drm_atomic_helper.c
index e737e45a3a702..66d223c2d9ab9 100644
--- a/drivers/gpu/drm/drm_atomic_helper.c
+++ b/drivers/gpu/drm/drm_atomic_helper.c
@@ -573,30 +573,6 @@ mode_valid(struct drm_atomic_state *state)
 	return 0;
 }
 
-static int drm_atomic_check_valid_clones(struct drm_atomic_state *state,
-					 struct drm_crtc *crtc)
-{
-	struct drm_encoder *drm_enc;
-	struct drm_crtc_state *crtc_state = drm_atomic_get_new_crtc_state(state,
-									  crtc);
-
-	drm_for_each_encoder_mask(drm_enc, crtc->dev, crtc_state->encoder_mask) {
-		if (!drm_enc->possible_clones) {
-			DRM_DEBUG("enc%d possible_clones is 0\n", drm_enc->base.id);
-			continue;
-		}
-
-		if ((crtc_state->encoder_mask & drm_enc->possible_clones) !=
-		    crtc_state->encoder_mask) {
-			DRM_DEBUG("crtc%d failed valid clone check for mask 0x%x\n",
-				  crtc->base.id, crtc_state->encoder_mask);
-			return -EINVAL;
-		}
-	}
-
-	return 0;
-}
-
 /**
  * drm_atomic_helper_check_modeset - validate state object for modeset changes
  * @dev: DRM device
@@ -768,10 +744,6 @@ drm_atomic_helper_check_modeset(struct drm_device *dev,
 		ret = drm_atomic_add_affected_planes(state, crtc);
 		if (ret != 0)
 			return ret;
-
-		ret = drm_atomic_check_valid_clones(state, crtc);
-		if (ret != 0)
-			return ret;
 	}
 
 	/*
